File: programmers/etc/level1/문자열_나누기.py
def solution(s):
    answer = 0
    pro, con = 0, 0
    for i in s:
        if pro == con:
            answer += 1
            target = i
        if i == target:
            pro += 1
        else:
            con += 1
    return answer


s = "banana"
print(solution(s)) # 3

s = "abracadabra"
print(solution(s)) # 6

s = "aaabbaccccabba"
print(solution(s)) # 3

File: Java/programmers/etc/level1/문자열나누기.java
package CodeTest.Java.programmers.etc.level1;

public class 문자열나누기 {
    public int solution(String st) {
        int answer = 0;
        int pro = 0, con = 0;
        String prev = "1";
        
        for (String s : st.split("")) {
            if (prev.equals("1")) {
                prev = s;
                pro++;
                answer++;
            } else if (prev.equals(s)) {
                pro++;
            } else {
                con++;
            }
            
            if (pro == con) {
                prev = "1";
                pro = 0; 
                con = 0;
            }
        }
                
        return answer;
    }
}
